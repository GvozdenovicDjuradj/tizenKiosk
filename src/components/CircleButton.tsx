import React from "react"
import {
  StyleProp,
  StyleSheet,
  Text,
  TextStyle,
  TouchableOpacity,
  TouchableOpacityProps,
} from "react-native"
import { NavigationEvents } from "react-navigation"

const circleSide =  260

const styles = StyleSheet.create({
  circleText: {
    fontSize: 30,
    textAlign: "center",
  },
  circleView: {
    alignItems: "center",
    borderRadius: circleSide / 2,
    elevation: 7,
    height: circleSide,
    justifyContent: "center",
    marginBottom: 10,
    padding: 5,
    shadowColor: "#000",
    shadowOffset: { width: 1, height: 1 },
    shadowOpacity: 0.8,
    shadowRadius: 10,
    width: circleSide,
  },
})

interface Props extends TouchableOpacityProps {
  text: string[];
  textStyle?: StyleProp<TextStyle>;
}

export default class extends React.Component<Props> {
  public state = {
    isDisabled: false
  }

  public disableHandler = (isDisabled: boolean) => () => {
    this.setState({
      isDisabled
    })
  }

  public render() {
    const { text, textStyle, style, ...otherProps } = this.props
    return (
      <React.Fragment>
        <NavigationEvents
          onWillFocus={this.disableHandler(false)}
          onWillBlur={this.disableHandler(true)}
        />
        <TouchableOpacity
          style={[styles.circleView, style]}
          disabled={this.state.isDisabled}
          {...otherProps}
        >
          {text.map((str: string, i: number) => {
            const txtStyle = Array.isArray(textStyle) ? textStyle[i] : textStyle
            return str ? (
              <Text key={i} style={[styles.circleText, txtStyle]}>{str}</Text>
            ) : null
          })}
        </TouchableOpacity>
      </React.Fragment>
    )
  }
}
