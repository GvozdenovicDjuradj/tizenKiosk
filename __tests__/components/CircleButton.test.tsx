import React from "react"
import renderer from "react-test-renderer"
import CircleButton from "../../src/components/CircleButton"

describe("Circle button test", () => {

  it("CircleButton should render successfully", () => {
    const text = ["Some", "text"]
    const wrapper = renderer.create(<CircleButton text={text} />)
    expect(wrapper.toJSON()).toMatchSnapshot()
  })

})
