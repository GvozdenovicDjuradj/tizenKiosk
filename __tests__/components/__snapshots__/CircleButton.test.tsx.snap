// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Circle button test CircleButton should render successfully 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderRadius": 130,
      "elevation": 7,
      "height": 260,
      "justifyContent": "center",
      "marginBottom": 10,
      "opacity": 1,
      "padding": 5,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 1,
        "width": 1,
      },
      "shadowOpacity": 0.8,
      "shadowRadius": 10,
      "width": 260,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontSize": 30,
          "textAlign": "center",
        },
        undefined,
      ]
    }
  >
    Some
  </Text>
  <Text
    style={
      Array [
        Object {
          "fontSize": 30,
          "textAlign": "center",
        },
        undefined,
      ]
    }
  >
    text
  </Text>
</View>
`;
